include/nbl/asset/IDescriptorSet.h

```diff
@@ -242,7 +242,7 @@ class NBL_API IEmulatedDescriptorSet
 			}
 
 			uint32_t offset;
-			E_DESCRIPTOR_TYPE descriptorType = EDT_INVALID;//whatever, default value
+			E_DESCRIPTOR_TYPE descriptorType = EDT_COUNT;//whatever, default value
```

Comment:
you could improve the `SBindingInfo` a little by tuning the `offset` into a 28 bit bitfield and the `descriptorType` into 4 bits

then static assert that the `EDT_COUNT<=15` and binding info is 4 bytes

Reply:
or by keeping `BindingInfo` in separate arrays (per descriptor type) you could get rid of the type field entirely (also makes any binary searches faster)

Reply:
Resolved.